src/test/java/hudson/scm/AbstractSubversionTest.java

```diff
@@ -61,6 +62,23 @@ protected Proc runSvnServe(File repo) throws Exception {
         } catch (IOException e) {
         	Assert.fail("Failed to launch svnserve. Do you have subversion installed?\n" + e);
         }
+
+        // If there is an already existing svnserve running on the machine
+        // We need to fail the build. We could change this to if the port is in use, listen to different port
+        Socket s = null;
+        try {
+          s = new Socket("localhost", 3690);
+          // If it gets this far, that means that it is able to send/receive information.
+          // Since the default svnserve port is currently in use, fail the build.
+          fail("Port 3690 is currently in use. Please kill anything running on this port and re-run.");
+        } catch (IOException e) {
+          // Port is not in use
+        } finally {
+          if (s != null) {
+            s.close();
+          }
+        }
+
```

Comment:
As noted in #161, this is wrong and causes test failures when you run with sufficient concurrency. Picking a different port would be fine if you were _using_ that port in the URL defined in the test case. But you are not. We just have for example `"svn://localhost/bob"` which assumes the default port of 3690.